File: brukerfeil.enode.frontend/src/types/ElementsMessage.ts
export type ElementsStatus =
    | 'Klar for sending'
    | 'Mottatt elektronisk'
    | 'Sendt'
    | 'Under sending'
    | 'Signert'
    | 'Levert'
    | 'Åpnet  av mottaker'
    | 'Overføring feilet'

type SendingStatus = {
    description: ElementsStatus
}

type SendingMethod = {
    description: string
}

type RegistryEntry = {
    id: number
    caseId: number
    documentNumber: number
    createdDate: Date
    lastUpdated: Date
    title: String
    senderRecipient: String
    isSenderRecipientRestricted: boolean
    case: Case
}

type Case = {
    id: number
    countOfRegistryEntries: number
    publicTitle: String
    caseNumber: String
    isRestricted: boolean
    createdDate: Date
    lastUpdated: Date
    caseDate: Date
}

export type ElementsMessage = {
    id: number
    name: string | null
    isPerson: boolean
    createdDate: Date
    lastUpdated: Date
    isRead: boolean
    conversationId: string
    externalId: string
    registryEntryId: number
    sendingMethod: SendingMethod | null
    sendingStatus: SendingStatus | null
    registryEntry: RegistryEntry | null
}

File: Brukerfeil.Enode/Brukerfeil.Enode.Common/Models/ElementsMessage.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace Brukerfeil.Enode.Common.Models
{
    public class ElementsMessageContent
    {
        public string odatacontext { get; set; }
        public IEnumerable<ElementsMessage> value { get; set; }
    }

    public class ElementsMessage
    {
        public int Id { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime LastUpdated { get; set; }
        public string ExternalId { get; set; }
        public string Name { get; set; }
        public bool IsPerson { get; set; }
        public int RegistryEntryId { get; set; }
        public bool IsRecipient { get; set; }
        public object IsRead { get; set; }
        public string ConversationId { get; set; }
        public SendingMethod SendingMethod { get; set; }
        public SendingStatus SendingStatus { get; set; }
        public RegistryEntry RegistryEntry { get; set; }
    }

    public class SendingMethod
    {
        public string Description { get; set; }
    }

    public class SendingStatus
    {
        public string Description { get; set; }
    }

    public class RegistryEntry
    {
        public int Id { get; set; }
        public int CaseId { get; set; }
        public int DocumentNumber { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime LastUpdated { get; set; }
        public String Title { get; set; }
        public String SenderRecipient { get; set; }
        public bool IsSenderRecipientRestricted { get; set; }
        public Case Case { get; set; }
    }

    public class Case
    {
        public int Id { get; set; }
        public int CountOfRegistryEntries { get; set; }
        public String PublicTitle { get; set; }
        public String CaseNumber { get; set; }
        public bool IsRestricted { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime LastUpdated { get; set; }
        public DateTime CaseDate { get; set; }
    }
}